azure/scope/managedcontrolplane.go

```diff
@@ -912,3 +914,35 @@ func (s *ManagedControlPlaneScope) PrivateEndpointSpecs() []azure.ASOResourceSpe
 func (s *ManagedControlPlaneScope) SetOIDCIssuerProfileStatus(oidc *infrav1.OIDCIssuerProfileStatus) {
 	s.ControlPlane.Status.OIDCIssuerProfile = oidc
 }
+
+// AKSExtension returns the cluster AKS extensions.
+func (s *ManagedControlPlaneScope) AKSExtension() []infrav1.MarketplaceExtension {
+	return s.ControlPlane.Spec.MarketplaceExtensions
+}
+
+// AKSExtensionSpecs returns the AKS extension specs.
+func (s *ManagedControlPlaneScope) AKSExtensionSpecs() []azure.ASOResourceSpecGetter[*asokubernetesconfigurationv1.Extension] {
+	if s.AKSExtension() == nil {
+		return nil
+	}
+	extensionSpecs := make([]azure.ASOResourceSpecGetter[*asokubernetesconfigurationv1.Extension], 0, len(s.ControlPlane.Spec.MarketplaceExtensions))
+	for _, extension := range s.AKSExtension() {
+		extensionSpec := &aksextensions.AKSExtensionSpec{
+			Name:                    extension.Name,
+			Namespace:               s.Cluster.Namespace,
```

Comment:
Sorry if this was asked earlier, why are we not allowing the user to configure the extension namespace?

Reply:
That's a mistake on my end. I think I erroneously assumed the extension has to be in the same namespace as the cluster. I'll add it as a field if it's fine to be in a different namespace.

Reply:
All of the other ASO resources are assumed to be in the cluster namespace currently and that would be a hard requirement for #4340, so I'd vote to keep this assumption for now.